### PR TITLE
Add a note about render-tests relying on the dev-build

### PR DESCRIPTION
**Pull request by smellyshovel**
In #4082 it was [concluded](https://github.com/maplibre/maplibre-gl-js/pull/4082#discussion_r1592606196) that sometimes (for certain render-tests) it's required to build the dev version of MapLibre. This PR proposes an addition to the CONTRIBUTING guide to reflect that.

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -13,8 +13,9 @@ Then run:
 npm test
 ```
 
-Note that for Linux you'll need to prefix test commands with `xvfb-run -a` for example `xvfb-run -a npm run test`
+Note that for Linux you'll need to prefix test commands with `xvfb-run -a` for example `xvfb-run -a npm run test`.
 Note that some tests rely on the built project and will not pass until you have run `npm run build-dist`.
+There are also some render tests that rely on the dev-build of the project, and they will not pass until you have run `npm run build-dev`.
```
Comment (HarelM) [inline test/README.md:18]: I believe all render tests require the dev build, don't they?

Comment (smellyshovel) [reply test/README.md:18]: @HarelM Oh, I'm not sure if *all* of them. I think when I was working on that other PR, some render tests were changing their results when I was rerunning `build-prod`. But I might be mistaken.

If you think that all of them do, I can change the wording.

Comment (HarelM) [reply test/README.md:18]: Il be surprised, see here: 
https://github.com/maplibre/maplibre-gl-js/blob/7936937526666e79f7b9173db0df7bfab85c0886/test/integration/render/run_render_tests.ts#L781

Comment (HarelM) [reply test/README.md:18]:
```suggestion
The render tests rely on the dev-build of the project, and they will not pass until you have run `npm run build-dev`. Don't forget to run it after changing your code as well.
```

**Review by HarelM — APPROVED** [review]

**Pull request merged**